File: backend/app/schemas/dataset.py
from datetime import datetime
from typing import List, Dict, Any, Optional
from pydantic import BaseModel, Field


class DatasetBase(BaseModel):
    name: str
    file_type: str
    file_size_bytes: int
    row_count: int
    column_count: int


class DatasetCreate(DatasetBase):
    storage_path: str


class DatasetResponse(DatasetBase):
    id: str
    created_at: datetime
    columns: List[str]
    column_types: Dict[str, str]

    class Config:
        from_attributes = True


class DatasetPreviewResponse(BaseModel):
    dataset_id: str
    columns: List[str]
    sample_data: List[Dict[str, Any]]
    total_rows: int

File: frontend/src/types/dataset.ts
export interface Dataset {
  id: string;
  name: string;
  file_type: string;
  file_size_bytes: number;
  row_count: number;
  column_count: number;
  created_at: string;
  columns: string[];
  column_types: Record<string, string>;
}

export interface ColumnSummary {
  name: string;
  data_type: string;
  semantic_type: 'numerical' | 'categorical' | 'datetime' | 'text' | 'id';
  missing_count: number;
  missing_percentage: number;
  unique_count: number;
}

export interface ProfileResponse {
  dataset_id: string;
  total_rows: number;
  total_columns: number;
  duplicate_rows: number;
  columns_summary: ColumnSummary[];
  quality_score: number;
}
